src/DataWrangling/ECCO/ECCODarwin.jl

```diff
@@ -0,0 +1,485 @@
+
+Download the dataset specified by `ECCODarwinMetadata`. If `ECCODarwinMetadata.dates` is a single date, 
+the dataset is downloaded directly. If `ECCODarwinMetadata.dates` is a vector of dates, each date
+is downloaded individually.
+The data download requires a username and password to be provided in the `ECCO_USERNAME` and
+`ECCO_PASSWORD` environment variables. This can be done by exporting the environment variables
+in the shell before running the script, or by launching julia with 
+
+```
+ECCO_USERNAME=myusername ECCO_PASSWORD=mypassword julia 
+```
+
+Arguments
+=========
+- `metadata::ECCODarwinMetadata`: The metadata specifying the dataset to be downloaded.
+"""
+function download_dataset(metadata::ECCODarwinMetadata; url = urls(metadata))
+    username = get(ENV, "ECCO_USERNAME", nothing)
+    password = get(ENV, "ECCO_PASSWORD", nothing)
+    dir = metadata.dir
+    
+    # Create a temporary directory to store the .netrc file
+    # The directory will be deleted after the download is complete
+    @root mktempdir(dir) do tmp
+
+        # Write down the username and password in a .netrc file
+        downloader = netrc_downloader(username, password, "ecco.jpl.nasa.gov", tmp)
+
+        asyncmap(metadata, ntasks=10) do metadatum # Distribute the download among tasks
+
+            fileurl  = metadata_url(url, metadatum) 
+            filepath = metadata_path(metadatum)
+
+            if !isfile(filepath)
+                instructions_msg = "\n See ClimaOcean.jl/src/ECCO/README.md for instructions."
+                if isnothing(username)
+                    msg = "Could not find the ECCO_PASSWORD environment variable. \
+                           See ClimaOcean.jl/src/ECCO/README.md for instructions on obtaining \
+                           and setting your ECCO_USERNAME and ECCO_PASSWORD." * instructions_msg
+                    throw(ArgumentError(msg))
+                elseif isnothing(password)
+                    msg = "Could not find the ECCO_PASSWORD environment variable. \
+                           See ClimaOcean.jl/src/ECCO/README.md for instructions on obtaining \
+                           and setting your ECCO_USERNAME and ECCO_PASSWORD." * instructions_msg
+                    throw(ArgumentError(msg))
+                end
+                # Get the data file
+                Downloads.download(fileurl, filepath; downloader, progress=download_progress)
+                # Get the meta file
+                Downloads.download(replace(fileurl, "data" => "meta"), 
+                                   replace(filepath, "data" => "meta"); 
+                                   downloader, 
+                                   progress=download_progress)
+            end
+        end
+    end
+    
+    return nothing
+end
+
+ECCODarwinMeshGrid(::ECCO4DarwinMonthly)   = GridSpec("LatLonCap", MeshArrays.GRID_LLC90)
+ECCODarwinMeshGrid(::ECCO270DarwinMonthly) = GridSpec("LatLonCap", MeshArrays.GRID_LLC90)
```

Comment:
Isn't the 270 version on the LLC270?

Reply:
Had to define LLC270 manually because there's no MeshArrays.GRID_LLC270